main.tf

```diff
@@ -56,19 +58,31 @@ resource "aws_kms_alias" "cluster" {
 resource "aws_eks_cluster" "default" {
   #bridgecrew:skip=BC_AWS_KUBERNETES_1:Allow permissive security group for public access, difficult to restrict without a VPN
   #bridgecrew:skip=BC_AWS_KUBERNETES_4:Let user decide on control plane logging, not necessary in non-production environments
-  count                         = local.enabled ? 1 : 0
-  name                          = module.label.id
-  tags                          = module.label.tags
-  role_arn                      = local.eks_service_role_arn
-  version                       = var.kubernetes_version
-  enabled_cluster_log_types     = var.enabled_cluster_log_types
-  bootstrap_self_managed_addons = var.bootstrap_self_managed_addons_enabled
+  count                     = local.enabled ? 1 : 0
+  name                      = module.label.id
+  tags                      = module.label.tags
+  role_arn                  = local.eks_service_role_arn
+  version                   = var.kubernetes_version
+  enabled_cluster_log_types = var.enabled_cluster_log_types
+  # Enabling EKS Auto Mode also requires that bootstrap_self_managed_addons is set to false
+  bootstrap_self_managed_addons = local.auto_mode_enabled ? false : var.bootstrap_self_managed_addons_enabled
```

Comment:
I consider it a bad idea to silently change input configuration like this.

Because this is a configuration conflict, and may change in the future, my first preference is to allow AWS to throw an error message rather than try to fix it, because then if AWS relaxes this condition, we don't need to make any changes.

An acceptable alternative is to add this check as a precondition and report an error.